packages/api/tests/lib/rules/test-rulesHelpers.js

```diff
@@ -1347,6 +1347,26 @@ test('Creating a rule trigger for a onetime rule succeeds', async (t) => {
   t.deepEqual(onetimeRule, rule);
 });
 
+test('Creating a rule trigger for a onetime rule with a DISABLED state is DISABLED', async (t) => {
+  const rule = fakeRuleFactoryV2({
+    workflow,
+    rule: {
+      type: 'onetime',
+    },
+    state: 'DISABLED',
+    meta: {
+      visibilityTimeout: 100,
+      retries: 4,
+    },
+  });
+
+  const invokeOneTimeSpy = sinon.spy(invoke);
+  const onetimeRule = await createRuleTrigger(rule);
+
+  t.false(invokeOneTimeSpy.called);
```

Comment:
Spies based on an include are a dangerous pattern , particularly in a non-serial test.     In general, as a pattern in core, we've tried to use something like 

```
createRuleTrigger(ruleItem, testContext)
```

where testContext sets a default but allows for a spy/stub/whatever to be passed in explicitly versus the meta modifications sinon provides. 

Perhaps more importantly, with 'something isn't called' it leads to situations where if you re-write this test as :



```
test.only('Creating a rule trigger for a onetime rule with a ENABLED state is ENABLED', async (t) => {
  const rule = fakeRuleFactoryV2({
    workflow,
    rule: {
      type: 'onetime',
    },
    state: 'ENABLED',
    meta: {
      visibilityTimeout: 100,
      retries: 4,
    },
  });

  const invokeOneTimeSpy = sinon.spy(invoke);
  const onetimeRule = await createRuleTrigger(rule);

  t.true(invokeOneTimeSpy.called);
  t.deepEqual(onetimeRule.state, 'ENABLED');
});
```

^^ this test fails because   t.true(invokeOneTimeSpy.called); is false.   You can't assign a spy like this, although you can use proxyquire....which should generally be avoided as well if possible, except in cases where modifying the call under test is infeasible. 

You can also validate this if you change the onetime block to 


```
  case 'onetime': {
    if (true) {
      await invoke(process.env.invoke, payload);
    }
    break;
  }
  ``` 
  
  this test still passes.

Reply:
I don't believe https://github.com/nasa/cumulus/pull/3371/commits/3f2a1beac5228c27b3d3477488e01766be7aafb1 removing the spy assertion resolves this - it's reduced the scope of the test.  

It is still valid, and needed, to validate that the rule not only is written to the database correctly, but *also* that the correct action is taken with regard to triggering the rule. 

This test doesn't evaluate the logic in 


```
  case 'onetime': {
    if (enabled) {
      await invoke(process.env.invoke, payload);
    }
    break;
  }
```

does the correct thing if disabled, nor the larger concern that is addressed with this ticket.  I can easily pass this test by modifying the logic entirely: 


 ```
  case 'onetime': {
    await invoke(process.env.invoke, payload);
    break;
  }
  ```
  
  As such, there's a hole in our testing around the intended changeset in this feature, namely in we don't ensure the rule isn't triggered when we set it to disabled.

Reply:
I agree that we need to capture testing the invocation of the rule upon creation